### ISSUE TITLE
Doubts regarding the code

### ISSUE THREAD

**Issue by Anirudh257** [tobyperrett/trx#4]
Thanks a lot for making your code public. I am trying to replicate your results on UCF-101 and HMDB. I have the following queries regarding them and the code in general:

1) I have a similar GPU configuration as you have reported on your Github(4 GPUs of 11GB) each but I am not able to run the code for either HMDB or UCF-101. When I use **resnet34** instead of **resnet50**, the sizes have decreased and I am able to run properly. But, does the accuracy get affected by doing this? My code for training is:

`python3 run.py -c checkpoint_dir --query_per_class 5 --shot 5 --way 5 --trans_linear_out_dim 1152 --test_iters 75000 --dataset ucf --split 3 -lr 0.001 --img_size 224 --scratch new --num_gpus 4 --method resnet34 -r --print_freq 1 --save_freq 1000
`

2) In the script, you have used confidence:

https://github.com/tobyperrett/trx/blob/390c20aa8ed7e309ae4949464fdcf015c4009329/run.py#L225

Why have you used **196.0**. I checked the paper but couldn't find the reason behind this. It will be great if you can explain the same. What is the reason behind the confidence metric?

3) A general doubt, when can I say that model has started overfitting in this context of the few-shot video classification? I know that evaluation is performed for 10000 tasks but in training, the default number of tasks is **100K**. Should I stop after I see that the losses have started approaching values of 0.0 decimal?

Thanks for clasrifying the validation part

**Comment by tobyperrett** [comment]
Hi - glad you've got something running!

1. I've only used ResNet50 on UCF and HMDB, but ResNet50 outperforms ResNet18 on the other datasets, so I would guess ResNet34 it wouldn't get as good results either. I'd recommend sticking with ResNet50, but having a lower --query_per_class (try 4 to start with). This might take longer to train, but should give similar results to what I got. If you are going to use ResNet34, I'd recommend playing around with the --trans_linear_out_dim as 1152 is greater than the feature dimensionality (512), which doesn't seem that intuitive to me.

2. Multiplying the standard deviation by 196 gives a 95% confidence interval (as the overall accuracy is the mean over 10,000 tasks).

3. You'd say that it's overfitting when performance starts to drop on the validation set. The 100,000 task default is what I checked up to for the OTAM SSv2 split, but you won't need to go that high for UCF and HMDB.

Hope that helps.
